### DISCUSSION TITLE
File not found error from dotnet-svcutil

### DISCUSSION THREAD

**Discussion by stnkblsm** [CoreWCF/CoreWCF#1198] [Q&A]
I have an old WCF service, and just finished porting it to .NET 6 (using CoreWCF).  The service seems to run fine and I can retrieve metadata over an http connection. I have been trying to use dotnet-svcutil 2.1.0 to update the service reference but have run into a few problems. I worked around most of them, but I have not been able to find a way around this one...

Microsoft (R) WCF Service Model Proxy Generation Tool for .Net Core platform
[Microsoft.Tools.ServiceModel.Svcutil, Version 2.1.0]
Copyright (c) Microsoft Corporation.  All rights reserved.

 

This tool collects information about how it is used in order to improve the tool. This functionality can be disabled by setting the environment variable "DOTNET_SVCUTIL_TELEMETRY_OPTOUT" to 1.

 

Resolving project references ...
Bootstrapping application ...
Attempting to download metadata from '[http://localhost:8020/EquipmentControlServer/mex?wsdl'](http://localhost:8020/EquipmentControlServer/mex?wsdl%27) using WS-Metadata Exchange and HttpGet.
Warning: Cannot import wsdl:portType
Detail: An exception was thrown while running a WSDL import extension: System.ServiceModel.Description.DataContractSerializerMessageContractImporter
Error: Could not load file or assembly 'System.Windows.Forms, Version=6.0.2.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'. The system cannot find the file specified.

I don't think there's anything in the service contract that requires this reference (I don't see anything in the WSDL file) but I suppose it's possible. I can't find this particular version of System.Windows.Forms on my computer (but there are many versions, and I haven't checked them all). I suspect it is the bootstrapper that requires this reference, but I don't know how to confirm that.

Has anyone else experience something like this?

**Comment by imcarolwang** [comment]
For debugging purposes, you can run "`set SVCUTIL_KEEP_BOOTSTRAPDIR=1`" before running dotnet-svcutil command line to generate the service proxy, after running the tool, go to the "%temp%/dotnet-svcutil-lib_Temp" folder and find the folder name with the latest timestamp, check the SvcutilBootstrapper project content for some clues.

If your service proxy has no common shared types that need to be reused, you can pass dotnet-svcutil with argument "`--noTypeReuse`" to suppress the bootstrapping process and hopefully the service proxy could be generated without error.

**Reply by stnkblsm** [reply]
Thanks Carol. I wasn't able to get the environment variable to work as you described, but with a little luck I was able to copy the contents of the temp folder before the files were deleted. I was able to build the bootstrapper and run it in the debugger, but it didn't provide any new clues.  The bootstrapper project file didn't contain anything unexpected.  We do need to reuse common shared types so the bootstrapping process seems to be required.

**Reply by imcarolwang** [reply]
Alternatively, try running the tool by specifying only the necessary reuse assemblies (with -r option) instead of reusing all referenced assembly types, this will reduce unnecessary dependencies for the bootstrapper project and possibly exclude System.Windows.Forms which may be a sub-dependency of one of them.
